core/src/main/java/org/apache/carbondata/core/util/ByteUtil.java

```diff
@@ -670,4 +671,23 @@ public static int putBytes(byte[] tgtBytes, int tgtOffset, byte[] srcBytes, int
     System.arraycopy(srcBytes, srcOffset, tgtBytes, tgtOffset, srcLength);
     return tgtOffset + srcLength;
   }
+
+  /**
+   * flatten the byte[][] to byte[] and return data after applying compression by compressor
+   * @param compressor compressor to use
+   * @return compressed data
+   */
+  public static byte[] flattenAndCompress(Compressor compressor, byte[][] byteArrayData) {
```

Comment:
Since it is ByteUtil, better use only flatten  not compress, compress can be done in caller method